homeassistant/components/whois/sensor.py

```diff
@@ -39,33 +41,52 @@ def setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the WHOIS sensor."""
-    domain = config[CONF_DOMAIN]
-    name = config[CONF_NAME]
-
+    LOGGER.warning(
+        "Configuration of the Whois platform in YAML is deprecated and will be "
+        "removed in Home Assistant 2022.4; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={CONF_NAME: config[CONF_NAME]},
```

Comment:
Do you not need to pass also the domain name?

Reply:
Ouch, nicely spotted! 🙏 thanks

Reply:
Added missing domain import in 90ada67